src/super_gradients/training/datasets/segmentation_datasets/segmentation_dataset.py

```diff
@@ -1,102 +1,55 @@
 import os
-import torch
-import random
-import numpy as np
-from tqdm import tqdm
-from typing import Callable
+from typing import Callable, Iterable
+
 import torchvision.transforms as transform
 from PIL import Image
+from tqdm import tqdm
 
 from super_gradients.common.decorators.factory_decorator import resolve_param
 from super_gradients.common.factories.transforms_factory import TransformsFactory
 from super_gradients.training.datasets.sg_dataset import DirectoryDataSet, ListDataset
-from super_gradients.training.transforms.transforms import RandomFlip, Rescale, RandomRescale, RandomRotate, \
-    CropImageAndMask, RandomGaussianBlur, PadShortToCropSize
 
 
 class SegmentationDataSet(DirectoryDataSet, ListDataset):
 
-    @resolve_param('image_mask_transforms', factory=TransformsFactory())
-    @resolve_param('image_mask_transforms_aug', factory=TransformsFactory())
+    @resolve_param('transforms', factory=TransformsFactory())
     def __init__(self, root: str, list_file: str = None, samples_sub_directory: str = None,
                  targets_sub_directory: str = None,
-                 img_size: int = 608, crop_size: int = 512, batch_size: int = 16, augment: bool = False,
-                 dataset_hyper_params: dict = None,
-                 cache_labels: bool = False, cache_images: bool = False, sample_loader: Callable = None,
-                 target_loader: Callable = None, collate_fn: Callable = None, target_extension: str = '.png',
-                 image_mask_transforms: transform.Compose = None, image_mask_transforms_aug: transform.Compose = None):
+                 cache_labels: bool = False, cache_images: bool = False,
```

Comment:
These are breaking changes which you did not take care of on the dataset_interface side.
It can't be that we changed the signature and nothing is changed in dataset_interfaces.py.
This might cause the current train_from_recipe to crash (and even worse- it might continue the training and  just takes some weird defaults).

Reply:
With the exception of the `transforms`, which were fixed in the latest commit for Cityscapes, nothing else was being used by any dataset that inherits from `SegmentationDataset`